tools/provision/chocolatey/thrift-dev.ps1

```diff
@@ -45,47 +49,38 @@ if (-not (Test-Path "$chocoBuildPath")) {
 }
 Set-Location $chocoBuildPath
 
-# Retrieve the source
-Invoke-WebRequest $url -OutFile "$packageName-$version.zip"
+# Retreive the source
+if (-not (Test-Path "$packageName-$version.zip")) {
+  Invoke-WebRequest $url -OutFile "$packageName-$version.zip"
+}
 
 # Extract the source
-7z x "$packageName-$version.zip"
-$sourceDir = "thrift-$version"
+$sourceDir = Join-Path $(Get-Location) "thrift-$version"
+if (-not (Test-Path $sourceDir)) {
+  $7z = (Get-Command '7z').Source
+  $7zargs = "x $packageName-$version.zip"
+  Start-OsqueryProcess $7z $7zargs
+}
 Set-Location $sourceDir
 
 # Thrift-dev requires this patch on windows, as our communications with the
 # thrift named pipe server happen to quickly, and we get loads of verbosity
 # this turns off said verbosity, as it's only concerned with our status
 # pings, and not the actual result flow of extensions itself.
-<#
-From 9fd916be17f221660e7af28ae3bfd47d0b846a46 Mon Sep 17 00:00:00 2001
-From: Nick Anderson <thor@fb.com>
-Date: Tue, 30 May 2017 16:21:46 -0700
-Subject: [PATCH 1/1] Patching out GLE ERROR_BROKEN_PIPE verbosity
-
----
- lib/cpp/src/thrift/windows/OverlappedSubmissionThread.cpp | 6 ++++--
- 1 file changed, 4 insertions(+), 2 deletions(-)
-
-diff --git a/lib/cpp/src/thrift/windows/OverlappedSubmissionThread.cpp b/lib/cpp/src/thrift/windows/OverlappedSubmissionThread.cpp
-index 5ac6fe00..0efdfbd2 100644
---- a/lib/cpp/src/thrift/windows/OverlappedSubmissionThread.cpp
-+++ b/lib/cpp/src/thrift/windows/OverlappedSubmissionThread.cpp
-@@ -52,8 +52,10 @@ uint32_t TOverlappedWorkItem::overlappedResults(bool signal_failure) {
-   BOOL result = ::GetOverlappedResult(h, &overlap, &bytes, TRUE);
-   if (signal_failure && !result) // get overlapped error case
-   {
--    GlobalOutput.perror("TPipe ::GetOverlappedResult errored GLE=", ::GetLastError());
--    throw TTransportException(TTransportException::UNKNOWN, "TPipe: GetOverlappedResult failed");
-+    if(!::GetLastError() == ERROR_BROKEN_PIPE){
-+      GlobalOutput.perror("TPipe ::GetOverlappedResult errored GLE=", ::GetLastError());
-+      throw TTransportException(TTransportException::UNKNOWN, "TPipe: GetOverlappedResult failed");
-+    }
-   }
-   return bytes;
- }
---
-2.12.0.windows.1
+$git = (Get-Command 'git').Source
+# Applying patches on Windows wasn't working until we were in an active repo
+
+<#Start-OsqueryProcess $git 'init'
```

Comment:
This is pretty neat - we have git patching for windows third party libs. The drawback here is the patch I'm attempting to use is currently failling our test runs, so I need to investigate how to better accomplish my goal.